roachtest: psycopg failed

roachtest.psycopg [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2898497&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2898497&tab=artifacts#/psycopg) on master @ [4dc05cceea254854317a9374a4b88df06c0946a6](https://github.com/cockroachdb/cockroach/commits/4dc05cceea254854317a9374a4b88df06c0946a6):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/psycopg/run_1
	orm_helpers.go:228,orm_helpers.go:154,psycopg.go:122,psycopg.go:135,test_runner.go:777: 
		Tests run on Cockroach v21.1.0-alpha.3-2478-g4dc05cceea
		Tests run against psycopg 2_8_6
		453 Total Tests Run
		450 tests passed
		3 tests failed
		319 tests skipped
		3 tests ignored
		0 tests passed unexpectedly
		1 test failed unexpectedly
		0 tests expected failed but skipped
		0 tests expected failed but not run
		---
		--- FAIL: tests.test_connection.TestEncryptPassword.test_encrypt_server (unexpected)
		For a full summary look at the psycopg artifacts 
		
		An updated blocklist (psycopgBlockList21_1) is available in the artifacts' psycopg log
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
# From https://go.crdb.dev/p/roachstress, perhaps edited lightly.
caffeinate ./roachstress.sh psycopg
```
</p>
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*psycopg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cmd/roachtest/psycopg_blocklist.go
+++ b/pkg/cmd/roachtest/psycopg_blocklist.go
@@ -29,8 +29,10 @@ var psycopgBlocklists = blocklistsForVersion{
 // After a failed run, an updated version of this blocklist should be available
 // in the test log.
 var psycopgBlockList21_1 = blocklist{
-	"tests.test_async_keyword.CancelTests.test_async_cancel":    "41335",
-	"tests.test_module.ExceptionsTestCase.test_9_6_diagnostics": "58035",
+	"tests.test_async_keyword.CancelTests.test_async_cancel": "41335",
+	// The following two items can be removed once there is a new psycopg2 release.
+	"tests.test_connection.TestEncryptPassword.test_encrypt_server": "42519",
+	"tests.test_module.ExceptionsTestCase.test_9_6_diagnostics":     "58035",
 }
 
 var psycopgBlockList20_2 = blocklist{

